### PR TITLE
PR template improvements 

### PR DESCRIPTION
**Pull request by sacha-l**
This PR updates the `PULL_REQUEST_TEMPLATE.md` to make it simpler to digest, with links to the labeling requirements and the style guide instead of duplicating that content. It also places more of a focus on the "Description" section, providing more granular guidelines to motivate authors to submit PRs with clear descriptions.

I've also made some minor edits to `CONTRIBUTING.adoc`, which I caught while making changes to the PR template.

### CODE REVIEW

**Comment by sam0x17** [comment]
The info about the labels is important though, no? I always find myself having to look at that

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,35 +2,31 @@
...
+
+Polkadot companion: (*if applicable*)
+
+Cumulus companion: (*if applicable*)
+
+# Checklist
+
+- [ ] My PR includes a detailed description as outlined in the "Description" section above
+- [ ] My PR follows the [labeling requirements](./CONTRIBUTING.adoc#merge-process) of this project (at minimum one label for each `A`, `B`, `C` and `D` required)
+- [ ] My code follows the [style guidelines](./STYLE_GUIDE.md) of this project
```
Comment (bkchr) [inline docs/PULL_REQUEST_TEMPLATE.md:25]: Why repeat stuff about the description again?

Labeling is enforced by CI and external people can not change lables?

Style guide aka fmt is also checked via CI

Comment (sacha-l) [reply docs/PULL_REQUEST_TEMPLATE.md:25]: How about just adding this note under the #Checklist heading:

> The CI bot will check that you have fulfilled the labeling and style guide requirements detailed in the checklist below. Don't worry if you don't have permissions to set the labels, you can still submit your PR and the person reviewing will set them for you.

**Comment by juangirini** [comment]
bot rebase

File: docs/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (bkchr) [reply docs/PULL_REQUEST_TEMPLATE.md:27]: Why put there redundant information? I mean people will see this ^^

Comment (kianenigma) [reply docs/PULL_REQUEST_TEMPLATE.md:27]: Yeah I would remove the old style-guide. It pertains to the days that we didn't use `rustfmt` 🙈

**Review by kianenigma — APPROVED** [review]
Please link to https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword as well, since I see a lot of people who seemingly are not aware of it.

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -2,35 +2,33 @@
...
```
Comment (sacha-l) [inline docs/PULL_REQUEST_TEMPLATE.md:27]:
```suggestion
```

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -2,35 +2,32 @@
...
-- [ ] **Docs:** You updated any rustdocs which may need to change.
-- [ ] **Polkadot Companion:** Has the PR altered the external API or interfaces used by Polkadot?
-  - [ ] If so, do you have the corresponding Polkadot PR ready?
-  - [ ] Optionally: Do you have a corresponding Cumulus PR?
-
-Refer to [the contributing guide](https://github.com/paritytech/substrate/blob/master/docs/CONTRIBUTING.adoc) for details.
-
-After you've read this notice feel free to remove it.
-Thank you!
+Thank you for your Pull Request! 🙏 Please make sure it follows the CONTRIBUTING guidelines outlined in [this document](./CONTRIBUTING.adoc) and fill out the sections below. Once you're ready to submit your PR for review, please delete this section and leave only the text under the "Description" heading.
```
Comment (sacha-l) [inline docs/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
Thank you for your Pull Request! 🙏 Please make sure it follows the contribution guidelines outlined in [this document](./CONTRIBUTING.adoc) and fill out the sections below. Once you're ready to submit your PR for review, please delete this section and leave only the text under the "Description" heading.
```

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -2,35 +2,32 @@
...
+Polkadot companion: (*if applicable*)
+
+Cumulus companion: (*if applicable*)
+
+# Checklist
+
+- [ ] My PR includes a detailed description as outlined in the "Description" section above
+- [ ] My PR follows the [labeling requirements](./CONTRIBUTING.adoc#merge-process) of this project (at minimum one label for each `A`, `B`, `C` and `D` required)
+- [ ] I have made corresponding changes to the documentation (if applicable)
+- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (sacha-l) [inline docs/PULL_REQUEST_TEMPLATE.md:28]:
```suggestion
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
```

File: docs/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -2,35 +2,32 @@
...
+Thank you for your Pull Request! 🙏 Please make sure it follows the contribution guidelines outlined in [this document](./CONTRIBUTING.adoc) and fill out the sections below. Once you're ready to submit your PR for review, please delete this section and leave only the text under the "Description" heading.
+
+# Description
+
+*Please include a summary of the changes and the related issue. Please also include relevant motivation and context, including:*
+
+- What does this PR do?
+- Why are these changes needed?
+- How were these changes implemented and what do they affect?
+
```
Comment (sacha-l) [inline docs/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion

*Use [Github semantic linking](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to address any open issues this PR relates to or closes.*

```

**Comment by sacha-l** [comment]
bot rebase

**Review by kianenigma — APPROVED** [review]

**Comment by sacha-l** [comment]
bot merge

**Pull request merged**